provider/packet/environ_network.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packet
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/names/v4"
+)
+
+var _ environs.Networking = (*environ)(nil)
+
+// Subnets returns basic information about subnets known by the provider for
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+
+	projectID := e.cloud.Credential.Attributes()["project-id"]
+	ips, _, err := e.packetClient.ProjectIPs.List(projectID, nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var projectSubnets []network.SubnetInfo
+	for _, ipblock := range ips {
+		subnetID, cidr, err := makeSubnetIDForNetwork("subnet-"+ipblock.ID, ipblock.Network, ipblock.CIDR)
```

Comment:
@achilleasa it seems no need to add prefix subnet as the function is already doing that

Reply:
```
  93.187.218.160/27:
    type: ipv4
    provider-id: subnet-subnet-3cc3db4a-47e2-4083-b19d-3b5533ecf216-93.187.218.160/27
```
changed it so it prints now:
```
 93.187.218.160/27:
    type: ipv4
    provider-id: subnet-3cc3db4a-47e2-4083-b19d-3b5533ecf216-93.187.218.160/27
    provider-network-id: 3cc3db4a-47e2-4083-b19d-3b5533ecf216
    status: in-use
    space: alpha
    zones: []
```